### ISSUE TITLE
Invalid SQL generated with cursor paging

### ISSUE THREAD

**Issue by andrewlutgen** [ChilliCream/hotchocolate#1218]
**Describe the bug**
A query with a sort and backwards paging is failing due to invalid SQL.

**To Reproduce**
1. Use Sort and Cursor Pagination
```
descriptor.Field(t => t.GetCompanies(default))
                .Authorize()
                .UsePaging<CompanyType>()
                .UseFiltering()
                .UseSorting();
```
2. Create a query with an order_by argument and backwards paging
```
query getCompanies {
  companies(
    last: 10
    before: "eyJfX3RvdGFsQ291bnQiOjMxLCJfX3Bvc2l0aW9uIjoyMH0="
    order_by: {
        name: ASC
    }
  ) {
    nodes {
      id
      name
    }
    pageInfo {
      endCursor
      hasNextPage
      hasPreviousPage
      startCursor
    }
    totalCount
  }
}
```
3. Observe error `Unexpected Execution Error`. Error does not occur with paging back to first page or when order_by is omitted. When error does occur, error message is:
`"message": "Invalid column name 'c'.\r\nThe column 'Name' was specified multiple times for 't'.\r\nThe column 'Name' was specified multiple times for 't0'."`
SQL profiler output generates the following related to above error:
```
exec sp_executesql N'SELECT COUNT(*)
FROM (
    SELECT [t].*
    FROM (
        SELECT TOP(@__p_0) [c].*, [c].[Name]
        FROM [Companies] AS [c]
        ORDER BY [c]
    ) AS [t]
    ORDER BY [t].[Name]
    OFFSET @__p_1 ROWS
) AS [t0]',N'@__p_0 int,@__p_1 int',@__p_0=20,@__p_1=10
```

**Expected behavior**
Cursor pagination to work forwards and backwards with and without sort.

**Desktop (please complete the following information):**
 - Hot Chocolate Version: 10.2.0
 - AspNetCore: 2.2

**Comment by michaelstaib** [comment]
First, thank you for reporting this. I have added this issue to 10.3.0 backlog.

**Comment by andrewlutgen** [comment]
Thanks @michaelstaib ! Any thoughts on when 10.3.0 is expected to release?

**Comment by michaelstaib** [comment]
We have to look if this will be even fixed in 10.3. have to evaluate what we can put really into 10.3. and what we will move to 10.4.

We are planing for end of next week.

**Comment by andrewlutgen** [comment]
In case anyone else comes runs into a similar issue, our work around was to keep track of the endCursor on page query and use forward paging instead. This worked for sequential forward and back paging.
